test/e2e/node_exporter_test.go

```diff
@@ -257,3 +257,17 @@ nodeExporter:
 	}
 
 }
+
+func TestNodeExporterGoMaxProcs(t *testing.T) {
+	t.Run("limited GOMAXPROCS", func(st *testing.T) {
+		f.PrometheusK8sClient.WaitForQueryReturn(
+			t, 5*time.Minute, `max(go_sched_gomaxprocs_threads{job="node-exporter"})`,
```

Comment:
we could add a metric with the number of CPU into the equation but it'll add complexity...